src/modules/folder/folder.usecase.ts

```diff
@@ -26,8 +26,11 @@ export class FolderUseCases {
     private readonly cryptoService: CryptoService,
   ) {}
 
-  async getFolder(folderId: FolderAttributes['id']) {
-    const folder = await this.folderRepository.findById(folderId);
+  async getFolder(
+    folderId: FolderAttributes['id'],
+    deleted: FolderAttributes['deleted'] = false,
```

Comment:
Having a boolean in a parameter without naming is a code smell, if you want to pass options to this method it is better advised to have it:

async getFolder(folderId, options);

Where options can have a {deleted: boolean} parameter.

Otherwise you can have someone use this method as: getFoler(<an id>, false); and someone new to the code will have to look inside to understand what is going on.

Better to have: 

getFolder(<an id>, {deleted: false}) so it is very clear what is going on.

Even though below you are calling it with an extra variable `deleted` , this behavior is not enforced and someone else might use it differently. 

I know having just one parameter is not such a big deal, but ultimately this can lead to having a method like:

getFolder(<an id>, false, true, false, false, true);

And then we have no idea what is going on, plus it's a source of bugs with the order